### ISSUE TITLE
Add Support for SSH for Gong OOB Connector

### ISSUE THREAD

**Issue by clay584** [trigger/trigger#197]
In order to connect to OOB terminal servers that run SSH, not telnet.

**Comment by coxley** [comment]
I'm also in favor of this. Don't have a clear image of the required work here, but will try to make time for investigation unless @jathanism has already thought about it.

Especially with OpenGear servers being pretty common, seems overdue.

**Comment by coxley** [comment]
I only have experience with one SSH terminal server (OpenGear), so I'm not sure the connection mechanic all of them use.

OpenGear uses ports:

`20xx/tcp`: Telnet lines
`30xx/tcp`: SSH lines
`40xx/tcp`: Raw TCP
`50xx/tcp`: RFC-2217

Implementing `30xx/tcp` via SSH would probably be the easiest, but also good to keep in mind OpenGear supports `[user]:[linenum]` set as the username.

Unsure if this would be desired or not, but might be interesting to check NetDevice for a certain parameter to determine if itself is a console server like OpenGear and SSH in with `[user]:serial` which brings up a menu of port numbers -> decription to select from. If other vendors don't implement this, though, probably a bit too one-sided to implement?

**Comment by mvh** [comment]
Cisco talks about 'reverse ssh' which is documented below:

http://www.cisco.com/c/en/us/td/docs/ios-xml/ios/sec_usr_ssh/configuration/15-mt/sec-usr-ssh-15-mt-book/sec-rev-ssh-enhanmt.html

I think it means that you have 'user:line' for the user, rather than just the user name, to connect to just that line.  I'll leave it to somebody better versed in Cisco-speak than me to decipher the above.

**Comment by bertjwregeer** [comment]
Avocent has their UMG's, they allow you to SSH directly to the console of a device with:

`user:portname`

That portname can change though, as an admin is allowed to give each console port its own name that identifies it.

If you ssh directly to the device, you get a menu, one of those menu items is a listing of connected ports, once there you can scroll up/down and select a console port.

**Comment by coxley** [comment]
Yeah so from all of this, it appears `user:portid` is a much more common implementation for SSH console servers and I'd vote it's what we should support.

Just a quick thought (but sounds reasonable):

* Implement an extra attribute role to tell if console is SSH vs Telnet
* Alter twister.py for username when connecting to such a device for the `OOBTerminalServerPort` value, letting the value be non-integer to support what @bertjwregeer mentioned (custom port desc/name).
